packages/eth-providers/src/base-provider.ts

```diff
@@ -546,17 +556,35 @@ export abstract class BaseProvider extends AbstractProvider {
       await this.api.isReadyOrError;
       await this.getNetwork();
 
-      if (!this.subscriptionStarted) {
-        this.subscriptionStarted = true;
-        await this.startSubscriptions();
+      if (!this.#subscription) {
+        this.#subscription = this.startSubscriptions();
```

Comment:
here I am thinking maybe we can unwrap `#subscription` from promise and do this one line
```
this.#subscription ??= await this.startSubscriptions();
```

Reply:
the reason I set the promise then wait is because if ready is called twice at the same time it won't call  this.startSubscriptions twice